@here/harp-examples/src/labelling_user-markers.ts

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    GeoCoordinates,
+    TileKey,
+    TilingScheme,
+    webMercatorTilingScheme
+} from "@here/harp-geoutils";
+import { MapControls, MapControlsUI } from "@here/harp-map-controls";
+import {
+    CopyrightElementHandler,
+    DataSource,
+    MapView,
+    TextElement,
+    Tile
+} from "@here/harp-mapview";
+import { FontUnit } from "@here/harp-text-canvas";
+import { assert } from "@here/harp-utils";
+import { VectorTileDataSource } from "@here/harp-vectortile-datasource";
+import * as THREE from "three";
+import { apikey } from "../config";
+
+const USER_MARKER_DATASOURCE = "user-markers-datasource";
+
+/**
+ * Basic example showing how to create a custom data source providing user text elements
+ * that change on arbitrary events.
+ */
+export namespace LabellingUserMarkers {
+    class UserMarkerDataSource extends DataSource {
```

Comment:
I think we should have a marker data source that can be reused and that maybe also adds some tiling for performance reasons.
@germanz also implemented a `MarkerDataSource` for JSAPI integration and it would be cool if they could be unified and made publicly available to harp users.